saleor/graphql/middleware.py

```diff
@@ -37,8 +37,14 @@ def user():
 class ChannelMiddleware:
     def resolve(self, next, root, info, **kwargs):
         request = info.context
-
-        channel_slug = kwargs.get("channel")
+        request_input = kwargs.get("input", {})
+        channel = kwargs.get("channel")
+        if isinstance(request_input, dict) and request_input:
+            channel_slug = request_input.get("channel")
+        elif channel is not None:
+            channel_slug = channel
+        else:
+            channel_slug = None
```

Comment:
Do we need this in this PR? I'm removing the middleware at all on my branch so this will be removed anyway.